Propagation.cpp

```diff
@@ -26,15 +26,33 @@ vector<Interval> MMP::prop_thru_interval(Vector2 src, Interval &w, Face *face,
             if (y < EPS)
                 y = 0;
 
-            new_intervals.push_back(Interval(x, y, info.e1, min(info.e2, edge->length()),
-                                             ps_d, face, edge, invert));
+            Interval ii(x, y, info.e1, min(info.e2, edge->length()),
+                        ps_d, face, edge, invert);
+
+            set_parent_iterator(ii, w);
+            new_intervals.push_back(ii);
         }
         // otherwise doesn't intersect this edge
     }
 
     return new_intervals;
 }
 
+void MMP::set_parent_iterator(Interval & ii, Interval & parent)
```

Comment:
@ahemanthkumar this will potentially lead to slow down. Can we have an interval storing it's own iterator when it's inserted? That way it will be easier to set the parent too.